tests/data/test_data_pipeline.py

```diff
@@ -0,0 +1,750 @@
+
+
+@pytest.mark.skipif(reason="Still using DataPipeline Old API")
+def test_datapipeline_transformations(tmpdir):
+
+    class CustomModel(Task):
+
+        def __init__(self):
+            super().__init__(model=torch.nn.Linear(1, 1), loss_fn=torch.nn.MSELoss())
+
+        def training_step(self, batch, batch_idx):
+            assert batch is None
+
+        def validation_step(self, batch, batch_idx):
+            assert batch is False
+
+        def test_step(self, batch, batch_idx):
+            assert len(batch) == 2
+            assert batch[0].shape == torch.Size([2, 1])
+
+        def predict_step(self, batch, batch_idx, dataloader_idx):
+            assert batch == [('a', 'a'), ('b', 'b')]
+            return torch.tensor([0, 0, 0])
+
+    class CustomDataModule(DataModule):
+
+        preprocess_cls = TestPreprocessTransformations
+
+    datamodule = CustomDataModule.from_load_data_inputs(1, 1, 1, 1, batch_size=2)
+
+    assert datamodule.train_dataloader().dataset[0] == (0, 1, 2, 3)
+    batch = next(iter(datamodule.train_dataloader()))
+    assert torch.equal(batch, torch.tensor([[0, 1, 2, 3, 5], [0, 1, 2, 3, 5]]))
+
+    assert datamodule.val_dataloader().dataset[0] == {'a': 0, 'b': 1}
+    assert datamodule.val_dataloader().dataset[1] == {'a': 1, 'b': 2}
+    with pytest.raises(MisconfigurationException, match="When ``per_sample_to_tensor_transform``"):
+        batch = next(iter(datamodule.val_dataloader()))
+
+    CustomDataModule.preprocess_cls = TestPreprocessTransformations2
+    datamodule = CustomDataModule.from_load_data_inputs(1, 1, 1, 1, batch_size=2)
+    batch = next(iter(datamodule.val_dataloader()))
+    assert torch.equal(batch["a"], torch.tensor([0, 1]))
+    assert torch.equal(batch["b"], torch.tensor([1, 2]))
+
+    model = CustomModel()
+    trainer = Trainer(
+        max_epochs=1,
+        limit_train_batches=2,
+        limit_val_batches=1,
+        limit_test_batches=2,
+        limit_predict_batches=2,
+        num_sanity_val_steps=1
+    )
+    trainer.fit(model, datamodule=datamodule)
+    trainer.test(model)
+    trainer.predict(model)
+
+    # todo (tchaton) resolve the lost reference.
+    preprocess = model._preprocess
+    # assert preprocess.train_load_data_called
+    # assert preprocess.train_per_sample_pre_tensor_transform_called
+    # assert preprocess.train_collate_called
+    assert preprocess.train_per_batch_transform_on_device_called
+    # assert preprocess.val_load_data_called
+    # assert preprocess.val_load_sample_called
+    # assert preprocess.val_per_sample_to_tensor_transform_called
+    # assert preprocess.val_collate_called
+    assert preprocess.val_per_batch_transform_on_device_called
+    # assert preprocess.test_load_data_called
+    # assert preprocess.test_per_sample_to_tensor_transform_called
+    # assert preprocess.test_per_sample_post_tensor_transform_called
+    # assert preprocess.predict_load_data_called
```

Comment:
if you do not use these assets, remove them

Reply:
Not, it is related to the previous bug. Just a reminder on how to update the test.